controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml

```diff
@@ -55,6 +55,14 @@ spec:
         - --enable-admission-plugins=NamespaceLifecycle,NodeRestriction,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota
         - --apiserver-count=1
         - --endpoint-reconciler-type=master-count
+        - --enable-aggregator-routing=true
+        - --requestheader-client-ca-file=/etc/kubernetes/pki/root/tls.crt
+        - --requestheader-allowed-names=""
+        - --requestheader-username-headers=X-Remote-User
+        - --requestheader-group-headers=X-Remote-Group
+        - --requestheader-extra-headers-prefix=X-Remote-Extra-
+        - --proxy-client-key-file=/etc/kubernetes/pki/apiserver/tls.key
+        - --proxy-client-cert-file=/etc/kubernetes/pki/apiserver/tls.crt
```

Comment:
It's not suggested to use the same CA and thus Cert/Key as the APIServer check out - https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/#ca-reusage-and-conflicts for more information, we actually already create a new CA/Cert/Key for "front-proxy" which is meant to be used for this purpose - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/controlplane/nested/controllers/nestedapiserver_controller.go#L228-L246 you'll need to add this to the volume mounts to add support for it.

Reply:
Done